common/workunit/workunit.cpp

```diff
@@ -13540,31 +13540,53 @@ void deleteK8sJob(const char *componentName, const char *job)
     jobname.toLowerCase();
     VStringBuffer deleteJob("kubectl delete job/%s", jobname.str());
     StringBuffer output, error;
-    bool ret = runExternalCommand(output, error, deleteJob.str(), nullptr);
+    bool ret = runExternalCommand(componentName, output, error, deleteJob.str(), nullptr);
     DBGLOG("kubectl delete output: %s", output.str());
     if (error.length())
         DBGLOG("kubectl delete error: %s", error.str());
     if (ret)
         throw makeStringException(0, "Failed to run kubectl delete");
 }
 
-void waitK8sJob(const char *componentName, const char *job, const char *condition)
+void waitK8sJob(const char *componentName, const char *job)
 {
     VStringBuffer jobname("%s-%s", componentName, job);
     jobname.toLowerCase();
+    VStringBuffer waitJob("kubectl get jobs %s -o jsonpath='{.status.active}'", jobname.str());
+    VStringBuffer getPhase("kubectl get pods --selector=job-name=%s --output=jsonpath='{.items[*].status.phase}'", jobname.str());
 
-    if (isEmptyString(condition))
-        condition = "condition=complete";
-
-    // MORE - blocks indefinitely here if you request too many resources
-    VStringBuffer waitJob("kubectl wait --for=%s --timeout=10h job/%s", condition, jobname.str());  // MORE - make timeout configurable
-    StringBuffer output, error;
-    bool ret = runExternalCommand(output, error, waitJob.str(), nullptr);
-    DBGLOG("kubectl wait output: %s", output.str());
-    if (error.length())
-        DBGLOG("kubectl wait error: %s", error.str());
-    if (ret)
-        throw makeStringException(0, "Failed to run kubectl wait");
+    unsigned delay = 100;
+    unsigned start = msTick();
+    unsigned pendingTimeout = 10000;
+    for (;;)
+    {
+        StringBuffer output, error;
+        bool ret = runExternalCommand(componentName, output, error, waitJob.str(), nullptr);
+        if (error.length())
+        {
+            DBGLOG("kubectl jobs error: %s", error.str());
+            break;
+        }
+        if (ret)
+            throw makeStringExceptionV(0, "Failed to run %s", waitJob.str());
+        if (!streq(output, "'1'"))
+        {
+            DBGLOG("kubectl jobs output: %s", output.str());
+            break;
+        }
+        ret = runExternalCommand(componentName, output.clear(), error.clear(), getPhase.str(), nullptr);
+        if (error.length())
+        {
+            DBGLOG("kubectl get pods error: %s", error.str());
+            break;
+        }
+        bool pending = streq(output, "'Pending'");
+        if (pending && msTick()-start > pendingTimeout)
+            throw makeStringExceptionV(0, "Failed to run %s - pod still pending after %d ms", jobname.str(), pendingTimeout);
+        MilliSleep(delay);
```

Comment:
If it's status is 'Running' , should it revert to using "kubectl wait --for .. " here?

Not sure it matters, but could spot status 'Success' here, and exit before delay and looping around.